.travis.yml

```diff
@@ -0,0 +1,20 @@
+sudo: required
+language: java
+dist: trusty
+jdk:
+- openjdk8
+install: true # skips travis' default installation step which executes gradle assemble.
+script: ./gradlew clean build --daemon
+# The before_cache and the cache steps cache the gradle installation on travis.
+before_cache:
+- rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+- rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
+  directories:
+  - $HOME/.gradle/caches/
+  - $HOME/.gradle/wrapper/
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
+  - ./gradlew shadowJar --daemon || travis_terminate 1
+  - ./gradlew dockerPushImage -Dbuild.version=$TRAVIS_COMMIT --daemon || travis_terminate 1
+  - ./gradlew dockerPushImage -Dbuild.version=$TRAVIS_BRANCH --daemon || travis_terminate 1
```

Comment:
What about created git tags/releases?

Reply:
If it's a tag/release TRAVIS_BRANCH will be the name of the tag. :


> `TRAVIS_BRANCH`
for push builds, or builds not triggered by a pull request, this is the name of the branch.
for builds triggered by a pull request this is the name of the branch targeted by the pull request.
for builds triggered by a tag, this is the same as the name of the tag (TRAVIS_TAG).